pkg/validation/interfaces/validator.go

```diff
@@ -44,3 +46,18 @@ func (validators Validators) Validate(objs ...interface{}) (results []errors.Man
 func (validators Validators) WithValidators(vals ...Validator) Validators {
 	return append(vals, validators...)
 }
+
+// DefinitionKey is a unqiue identifier for an API definition.
+type DefinitionKey struct {
```

Comment:
Is this type strictly necessary? Can we just stringify the gvk and compare that directly, rather than adding this type definition just to include the `String()` function?

Reply:
That would work although I don't like breaking the semantics of a GVK just for a string. Also I could see this being used elsewhere in the future, like its counterpart in [operator-registry](https://github.com/operator-framework/operator-registry/blob/v1.12.4/pkg/registry/types.go#L35).

Reply:
Removed.